### PR TITLE
Remove references to infer_schema macro from the docs

### LINKED ISSUE #2281: Docs: infer_schema is deprecated
[This line](https://github.com/diesel-rs/diesel/blob/master/diesel/src/lib.rs#L18) should be updated to reflect the deprecation of `infer_schema` [as of 1.3.0](https://github.com/diesel-rs/diesel/blob/v1.3.0/CHANGELOG.md#deprecated).

### PR DESCRIPTION
**Pull request by blasrodri**
Fixes #2281

### CODE REVIEW

File: diesel/src/macros/mod.rs [hunk 1]
```diff
@@ -891,8 +891,8 @@ macro_rules! __diesel_table_query_source_impl {
 /// The generated `ON` clause will always join to the primary key of the parent
 /// table. This macro removes the need to call [`.on`] explicitly, you will
 /// still need to invoke [`allow_tables_to_appear_in_same_query!`] for these two tables to
-/// be able to use the resulting query, unless you are using `infer_schema!` or
-/// `diesel print-schema` which will generate it for you.
+/// be able to use the resulting query, unless you are using `diesel print-schema`
```
Comment (jdlambert) [inline diesel/src/macros/mod.rs]: A comma seems appropriate here, before the "which":

> ...unless you are using `diesel print-schema`, which will generate it for you.

File: diesel/src/macros/mod.rs [hunk 2]
```diff
@@ -891,8 +891,8 @@ macro_rules! __diesel_table_query_source_impl {
...
+/// which will generate it for you.
```
Comment (jdlambert) [inline diesel/src/macros/mod.rs]: It's not immediately apparent to me what this "it" refers to.

Comment (blasrodri) [reply diesel/src/macros/mod.rs]: I agree. And it's not even clear to me either.

Comment (weiznich) [reply diesel/src/macros/mod.rs]: It refers to `allow_tables_to_appear_in_same_query!.

Comment (blasrodri) [reply diesel/src/macros/mod.rs]: @weiznich Should I rephrase it, or leave it as it is then?

Comment (weiznich) [reply diesel/src/macros/mod.rs]: If you have an idea how to rephrase it, then yes. Otherwise it is not required for merging this pr.

**Pull request merged**